### PR TITLE
Create Azure Blob Storage export documentation

### PR DESCRIPTION
**Pull request by LeontiBrechko**
# Amplitude Developer Docs PR

## Description

Adds Azure Blob Storage export documentation

## Deadline

Feb 16, 2024

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [x] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/destinations/azure-blob-storage.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+---
+title: Send Amplitude Event Data to Azure Blob Storage
+description: Export Amplitude event data and merged user data to your Azure Blob Storage container.
+---
+
+Amplitude users can now export Amplitude event data and merged user data to their Azure Blob Storage container. Azure Blob Storage's container policies allow you to manage and programmatically export this data into an Azure container. Using the Amplitude UI, you can set up recurring syncs as often as once per hour.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:6]:
```suggestion
Amplitude's Azure Blob Storage destination enables you to export Amplitude event data and merged user data to your Azure Blob Storage container. You can use Azure Blob Storage's container policies to manage and export this data programmatically into an Azure container. Amplitude allows recurring syncs as often as once per hour.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+--8<-- "includes/editions-all-editions.md"
+
+## Azure Prerequisites for Amplitude Data Export
+
+Before you can export data from Amplitude to Azure Blob Storage, there are several prerequisites that need to be met in your Azure environment.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:11]:
```suggestion
Before you can export data from Amplitude to Azure Blob Storage, ensure your Azure environment meets the following prerequisites:
```

File: docs/data/destinations/azure-blob-storage.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+1. **Set Up Azure Storage Account and Container:**
+
+    If you haven't already set up an Azure Storage Account and a Blob Storage container, follow these guides:
+
+    - [Create an Azure Storage Account](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create)
+    - [Create a container in Azure Blob Storage](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-portal)
+
+2. **Create an Azure Service Principal**:
+
+    Create a service principal in Azure for Amplitude to use when accessing your Blob Storage. Take note of the `tenantId`, `clientId`, and `clientSecret`, as these will be needed for setting up the connection in Amplitude. Follow this guide for creating a service principal:
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:22]:
```suggestion
    Create a service principal in Azure for Amplitude to use to access your Blob Storage. Copy the `tenantId`, `clientId`, and `clientSecret`, Amplitude requires these details to set up the connection with Azure. Follow this guide to create a service principal:
```

File: docs/data/destinations/azure-blob-storage.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+    - [Create a service principal in Azure](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+3. **Grant Required Permissions to Service Principal**:
+
+    Assign the necessary permissions for Azure Blob Storage to your service principal:
+
+    !!!note
+       
+        Read permissions required for ensuring data exported only once during recurring export. Delete permissions required for deduplication during manual export (a.k.a. backfill export).
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:32]: @LeontiBrechko I don't quite follow this note. Is this accurate?

Azure requires `read` permissions to ensure data isn't exported more than once for recurring exports. 
Azure requires  `delete` permissions to enable deduplication during a manual export, for example when you export backfill data.

File: docs/data/destinations/azure-blob-storage.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    - Amplitude EU IP addresses:
+        - `3.124.22.25`
+        - `18.157.59.125`
+        - `18.192.47.195`
+
+    For guidance on modifying these settings, see:
+
+    - [Configure Azure Storage firewalls and virtual networks](https://docs.microsoft.com/en-us/azure/storage/common/storage-network-security)
+
+Completing these steps ensures that your Azure environment is properly configured and ready to securely receive data exports from Amplitude.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:60]:
```suggestion
Once complete, these steps ensure your Azure environment is ready to receive secure data exports from Amplitude.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+
+## Set up a Recurring Data Export to Azure Blob Storage
+
+To export your Amplitude data to Azure Blob Storage, follow these steps:
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:65]:
```suggestion
To export your Amplitude data to Azure Blob Storage:
```

File: docs/data/destinations/azure-blob-storage.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+1. In Amplitude, navigate to **Data -> Manage -> Destinations -> Add Destination**.
+2. Select **Azure Blob Storage - Events & Merged Users**.
+3. On the **Getting Started** tab, select the data you'd like to export. Options include *Export events ingested today and moving forward*, *Export all merged Amplitude IDs*, or both.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:69]:
```suggestion
3. On the **Getting Started** tab, select the data to export. Options include *Export events ingested today and moving forward*, *Export all merged Amplitude IDs*, or both.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+    !!!note
+
+        You can export these two different data types to separate containers, if you prefer. You'll just need to complete the setup flow twice: once for each data type.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:73]: @LeontiBrechko 
Does each data type / container require a separate instance of the destination?

File: docs/data/destinations/azure-blob-storage.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+4. Review the Event table and Merge IDs table schemas and click **Next**.
+5. In the *Set Up Credentials* section, enter the `tenantId`, `clientId`, and `clientSecret` from the service principal you created.
+6. After entering the credentials, fill out the Azure Blob Storage container details in the *Azure Blob Storage Container Details* section.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:77]:
```suggestion
6. Complete the Azure Blob Storage container details in the *Azure Blob Storage Container Details* section. Click **Next**.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+7. Click **Next**. Amplitude attempts a test upload to check that the entered credentials work. If the upload is successful, click **Finish** to complete the Azure Blob Storage destination configuration and activation.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:78]:
```suggestion
7. Amplitude attempts a test upload to check that the entered credentials work. If the upload is successful, click **Finish** to complete the Azure Blob Storage destination configuration and activation.
```

If the test upload isn't successful, how detailed are the error messages? Should we include troubleshooting information here?

File: docs/data/destinations/azure-blob-storage.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+All future events/merged users are automatically sent to your Azure Blob Storage. Amplitude exports files to your Azure account every hour.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:80]:
```suggestion
When complete, Amplitude sends all future events/merged users to your Azure Blob Storage once per hour.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+## Run a manual export
+
+You can backfill historical data to Azure Blob Storage by manually exporting data.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:84]:
```suggestion
Complete a manual export to backfill and send historical data to Azure Blob Storage.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+
+1. Go to the Azure Blob Storage export connection page created in the section above.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:86]:
```suggestion
1. Open the instance of the Azure Blob Storage destination you created above. Navigate to the Export Connection page.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+2. Go to **Backfills** tab.
+3. Select the desired date range.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:88]:
```suggestion
3. Select the date range to export.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+4. Click **Start Backfill**.
+
+If the backfill range overlaps with the range of previously exported data, Amplitude will de-duplicate overlapping data.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:91]:
```suggestion
If the backfill range overlaps with the range of previously exported data, Amplitude de-duplicates any overlapping data.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+
+**![screenshot of the export data modal](../../assets/images/integrations-gcs-export-manual-export.png)**
+
+## Exported data format
+
+### Raw event file and data format
+
+Data is exported hourly as zipped archive JSON files, and partitioned by the hour with one or multiple files per hour. Each file contains one event JSON object per line.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:99]:
```suggestion
Amplitude exports data hourly as a zipped archive of JSON files, partitioned by the hour with one or more files per hour. Each file contains one event JSON object per line.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+
+File names have the following syntax, where the time represents when the data was uploaded to Amplitude servers in UTC (for example, `server_upload_time`):
+
+`projectID_yyyy-MM-dd_H#partitionInteger.json.gz`
+
+For example, the first partition of data uploaded to this project, on Jan 25, 2020, between 5 PM and 6 PM UTC, is in the file:
+
+`187520_2020-01-25_17#1.json.gz`
+
+Here is the exported data JSON object schema:
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:109]:
```suggestion
The exported data JSON object schema is:
```

File: docs/data/destinations/azure-blob-storage.md [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+   "device_family": string,
+   "sample_rate": null,
+   "idfa": string,
+   "client_event_time": UTC ISO-8601 timestamp,
+}
+```
+
+### Merged Amplitude IDs file and data format
+
+Data is exported hourly as zipped archive JSON files. Each file contains one merged Amplitude ID JSON object per line.
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:167]:
```suggestion
Amplitude exports data as a zipped archive of JSON files. Each file contains one merged Amplitude ID JSON object per line.
```

File: docs/data/destinations/azure-blob-storage.md [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+
+File names have the following syntax, where the time represents when the data was uploaded to Amplitude servers in UTC (for example `server_upload_time`):
+
+`-OrgID_yyyy-MM-dd_H.json.gz`
+
+For example, data uploaded to this project, on Jan 25, 2020, between 5 PM and 6 PM UTC, is in the file:
+
+`-189524_2020-01-25_17.json.gz`
+
+Merged ID JSON objects have the following schema:
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:177]:
```suggestion
The exported data JSON object schema is:
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

File: docs/data/destinations/azure-blob-storage.md (hunk 4 above)
Comment (LeontiBrechko) [reply docs/data/destinations/azure-blob-storage.md:32]: ~~Azure~~ Amplitude requires read permissions to ensure data isn't exported more than once for recurring exports.
~~Azure~~ Amplitude requires delete permissions to enable deduplication during a manual export, for example when you export backfill data.

File: docs/data/destinations/azure-blob-storage.md [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+
+    - [Create a service principal in Azure](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+3. **Grant Required Permissions to Service Principal**:
+
+    Assign the necessary permissions for Azure Blob Storage to your service principal:
+
+    !!!note
+       
+        Read permissions required for ensuring data exported only once during recurring export. Delete permissions required for deduplication during manual export (a.k.a. backfill export).
```
Comment (markzegarelli) [inline docs/data/destinations/azure-blob-storage.md:32]:
```suggestion
        Amplitude requires the following permissions in Azure:
        - `read` to ensure data isn't exported more than once for recurring exports.
        - `delete` to enable deduplication during a manual export, for example when you export backfill data.
```

**Review by markzegarelli — APPROVED** [review]

**Comment by LeontiBrechko** [comment]
@markzegarelli thanks a lot for your input! 

Just a not for myself: we need to add the Azure link to this page as well: https://pr-171.d19s7xzcva2mw3.amplifyapp.com/data/destination-cloud-storage-overview/

**Comment by markzegarelli** [comment]
> https://pr-171.d19s7xzcva2mw3.amplifyapp.com/data/destination-cloud-storage-overview/

@LeontiBrechko I added the icon and updated the page, but I'm not sure if it's the right logo. Please let me know once the preview build completes.

**Pull request merged**